app/scripts/metamask-controller.js

```diff
@@ -3895,7 +3896,7 @@ export default class MetamaskController extends EventEmitter {
     const pendingNonce = nonceDetails.params.highestSuggested;
 
     releaseLock();
-    return pendingNonce;
+    return bnToHex(new BN(pendingNonce));
```

Comment:
Can this be replaced with

```suggestion
    return `0x${BigInt(pendingNonce).toString(16)}`
```

This uses the browsers inbuilt bigint.

This is in line with your discussion about reducing build size cc @danjm